### PR TITLE
Admin panel

### PR DESCRIPTION
**Pull request by Sciguymjm**

### CODE REVIEW

File: webapp/src/Components/Player/Overview/SideBar/PlayerProfilePicture.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            4: "Developer"
+        }
+        const hierarchy = [1, 4, 2, 3]
+        let userRole = 0
+        for (const role of hierarchy) {
+            if (groups.indexOf(role) !== -1) {
+                userRole = role
+                break
+            }
+        }
```
Comment (Abbondanzo) [inline webapp/src/Components/Player/Overview/SideBar/PlayerProfilePicture.tsx]:
```javascript
const groups = [1]

const roles = [
  {
    name: "Admin",
    color: "red",
    letter: "A",
    hierarchy: 1
  },
  {
    name: "Developer",
    color: "#03A9F4",
    letter: "D",
    hierarchy: 2
  },
  {
    name: "Alpha Tester",
    color: "#FFC107",
    letter: "α",
    hierarchy: 3
  },
  {
    name: "Beta Tester",
    color: "#8BC34A",
    letter: "β",
    hierarchy: 4
  }
]

const userRole = roles
  .filter(({ hierarchy }) => groups.includes(hierarchy))
  .sort(({ hierarchy: a }, { hierarchy: b }) => a - b)[0]

```

Comment (Abbondanzo) [reply webapp/src/Components/Player/Overview/SideBar/PlayerProfilePicture.tsx]: Probably doesn't even need `hierarchy`. Could rename that in each of the objects to ID and then assume that hierarchy is determined by the first element of the filtered list

Comment (Abbondanzo) [reply webapp/src/Components/Player/Overview/SideBar/PlayerProfilePicture.tsx]:
```javascript
const roles = [
  {
    id: 1,
    name: "Admin",
    color: "red",
    letter: "A"
  },
  {
    id: 2,
    name: "Developer",
    color: "#03A9F4",
    letter: "D"
  },
  {
    id: 3,
    name: "Alpha Tester",
    color: "#FFC107",
    letter: "α"
  },
  {
    id: 4,
    name: "Beta Tester",
    color: "#8BC34A",
    letter: "β"
  }
]

const userRole = roles.filter(({ id }) => groups.includes(id))[0]
```

File: backend/blueprints/spa_api/service_layers/admin.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from backend.utils.checks import is_admin
+
+
+class AdminPanelHandler:
+
+    @staticmethod
+    @with_session
+    def add_group_to_user(id_: str, group: int, session=None):
+        if not is_admin():  # putting this here so that it *can't* be run without being admin from anywhere
+            raise CalculatedError(401, "Operation not permitted")
```
Comment (dtracers) [inline backend/blueprints/spa_api/service_layers/admin.py]: AuthorizationException

File: backend/blueprints/spa_api/service_layers/admin.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        player = session.query(Player).filter(Player.platformid == id_).one_or_none()
+        if player is None:
+            raise CalculatedError(400, "Player does not exist")
```
Comment (dtracers) [inline backend/blueprints/spa_api/service_layers/admin.py]: PlayerNotFound

File: backend/blueprints/spa_api/service_layers/admin.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        flag_modified(player, "groups")
+        session.add(player)
+        session.commit()
+        return jsonify({"status": "Success"})
+
+    @staticmethod
+    @with_session
+    def remove_group_from_user(id_: str, group: int, session=None):
+        if not is_admin():  # putting this here so that it *can't* be run without being admin from anywhere
+            raise CalculatedError(401, "Operation not permitted")
```
Comment (dtracers) [inline backend/blueprints/spa_api/service_layers/admin.py]: AuthorizationException

File: backend/blueprints/spa_api/service_layers/admin.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+        player = session.query(Player).filter(Player.platformid == id_).one_or_none()
+        if player is None:
+            raise CalculatedError(400, "Player does not exist")
```
Comment (dtracers) [inline backend/blueprints/spa_api/service_layers/admin.py]: PlayerNotFound

File: backend/utils/logging.py [hunk 6]
```diff
@@ -36,3 +42,65 @@ def log_error(exception: Exception, message: str = None, logger: logging.Logger
...
+                game_hash = proto_game.game_metadata.id
+        log = ReplayLog(uuid=replay_uuid, result=ReplayResult.SUCCESS, error_type="",
+                        log="", params=str(query_params), game=game_hash)
+        session.add(log)
+        session.commit()
+
+    @staticmethod
+    @with_session
+    def get_logs(page, limit, search, session=None):
+        # if is_admin():
```
Comment (dtracers) [inline backend/utils/logging.py]: add admin

**Pull request merged**